src/test/java/com/cegeka/academy/repository/UserChallengeTest.java

```diff
@@ -48,6 +50,32 @@ public void init() {
         challenge.setStatus("Active");
         challenge.setCreator(user);
         challengeRepository.save(challenge);
+
+        challengeAnswer.setImagePath("imagePath");
+        challengeAnswer.setVideoAt("videoAt");
+        challengeAnswer.setAnswer("answer");
+        challengeAnswerRepository.save(challengeAnswer);
+    }
+
+    @AfterEach
+    public void destroy(){
+
+        if(user != null) {
+            userRepository.delete(user);
```

Comment:
Eu optam pentru delete all. In felul asta cred ca te poti asigura ca daca mai adaugi ceva in baza de date in timpul unui test, nu ramane acolo pentru urmatoarele.

Reply:
Faza este ca nush daca ar fi corect sa stergi tot ce este in baza de date, ci doar ce ai folosit tu pentru teste. In tabela User, de ex, sunt inserate default trei inregistrari si nu cred ca ar trebui sterse.

Reply:
Eu i-am tot sters si nu s-a intamplat nimic. Ideea e ca daca stergi unul cate unul, poate ai un test unde salvezi un user si uiti sa-l stergi si chestia asta poate sa afecteze un alt test.

Reply:
Nu am zis ca s-ar intampla ceva rau daca stergi tot din tabela User. Ce vreau eu sa zic este ca ar trebui sa stergem doar ce folosim noi in teste si sa nu ne legam de ceea ce exista.

Reply:
In plus, incearca sa salvezi intr-un test o inregistrare si o accesezi intr-un alt test si vei vedea ca nu se pastreaza. Dupa cum a spus Bianca la inceput, pentru teste se foloseste baza de date in-memory H2 care mentine informatii doar pe timpul rularii fiecarui test.

Reply:
Pai si in cazul asta mai are rost sa avem o metoda de destroy() daca oricum nu persista? Eu am senzatia ca persista de la un test la altul cand rulezi toate testele dintr-o clasa. Nu persista in schimb daca rulezi clasa acum, pentru rularile viitoare.

Reply:
Hai ca fac un experiment si las aici concluzia.

Reply:
Persista de la un test la altul

![Screenshot from 2019-08-18 16-46-42](https://user-images.githubusercontent.com/43818861/63225381-e4aa3480-c1d7-11e9-8fa0-8446ab2d60be.png)

Reply:
La mine nu se pastreaza. Mi se pare foarte ciudat.
<img width="1440" alt="Screenshot 2019-08-18 at 17 06 46" src="https://user-images.githubusercontent.com/43901377/63225646-f17c5780-c1da-11e9-87cc-7bfecc778995.png">

Reply:
In MySql ai cumva salvat userul ala pe care faci tu testul?

Reply:
Ok. Asta e super dubios :))

Reply:
Nu e nimic salvat inainte

Reply:
Nush ce se intampla. Am sa modific cu deleteAll(), dar nush daca e ok

Reply:
Gata. Stiu de ce. Clasa are @Transactional sus pe cand a mea nu avea

Reply:
da, ai dreptate

Reply:
@Transactional face ca user-ul sa fie valabil doar in cadrul testului in care este salvat.

Reply:
Ca sa scapam de problema asta, ca raman inregistrarile salvate de la un test la altul sau nu, am sa fac deleteAll() si aia e. Poate nu toti folosesc adontarea transactional si ajungem in situatia despre care vorbeai mai sus.

Reply:
Am mai facut niste teste. Concluziile sunt: Cu `@Transactional` nu persista nimic din ceea ce este rulat in teste, dar persista ceea ce a fost salvat in @BeforeEach. Inseamna ca daca folosim adnotarea, trebuie in `@AfterEach` sa stergem ce a fost adaugat in @BeforeEach, iar de restul se ocupa programul. Daca nu il folosim, pe `@AfterEach` trebuie sa stergem si ce s-a adaugat in teste.

Reply:
Da, asa e. Dar nush cum ar trebui mai exact sa lasam. Ori stergem tot, ori doar ce folosim.